python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -84,11 +84,11 @@ def __init__(
         auto_materialize_policies_by_key: Mapping[AssetKey, Optional[AutoMaterializePolicy]],
         backfill_policies_by_key: Mapping[AssetKey, Optional[BackfillPolicy]],
         code_versions_by_key: Mapping[AssetKey, Optional[str]],
-        is_observable_by_key: Mapping[AssetKey, bool],
         auto_observe_interval_minutes_by_key: Mapping[AssetKey, Optional[float]],
         required_assets_and_checks_by_key: Mapping[
             AssetKeyOrCheckKey, AbstractSet[AssetKeyOrCheckKey]
         ],
+        execution_types_by_key: Mapping[AssetKey, Sequence[AssetExecutionType]],
```

Comment:
We use a `Sequence` because in the `ExternalAssetGraph`, it is possible to have assets that are both observable and materializable due to definitions in different code locations.

Reply:
This is really confusing. Is there a way that we can canonicalize this at this level?